src/Solvers/multigrid_solver.jl

```diff
@@ -128,30 +163,84 @@ function fill_matrix_elements!(A, template_field, linear_operator!, args...)
     return nothing
 end
 
+function fill_matrix_elements2!(template_field, linear_operator!, args...)
+    Nx, Ny, Nz = size(template_field)
+    make_column(f) = reshape(interior(f), Nx*Ny*Nz)
+
+    eᵢⱼₖ = similar(template_field)
+    ∇²eᵢⱼₖ = similar(template_field)
+
+    arch = architecture(template_field)
+    colptr = array_type(arch){Int64}(undef, Nx*Ny*Nz+1)
+    rowval = array_type(arch){Int64}(undef, 0)
```

Comment:
```suggestion
    colptr = array_type(arch){Int}(undef, Nx*Ny*Nz+1)
    rowval = array_type(arch){Int}(undef, 0)
```

`Int` is either `Int64` or `Int32` depending on the machine you are on. Probably better idea?